### PR TITLE
THC SELECT

### PR DESCRIPTION
**Pull request by fdmalone**
Adds a somewhat incomplete SELECT bloq for the molecular THC hamiltonian (See #386 for deficiencies).

My plan is to provide primitive bloqs this week for the molecular 2nd quantized Hamiltonians and I will follow up with more detailed implementations in time.

### CODE REVIEW

**Comment by fdmalone** [comment]
#350 should be merged first

File: qualtran/bloqs/chemistry/thc_select.py [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+
+    Args:
+        num_mu: THC auxiliary index dimension $M$
+        num_spin_orb: number of spin orbitals $N$
+        angles: list of num_mu + num_spin_orb / 2 lists of rotation angles each
+            of length num_spin_orb//2. Each rotation angle should be a
+            `num_bits_theta` fixed point approximation to the angle.
+
+    Registers:
+     - mu: $\mu$ register.
```
Comment (mpharrigan) [inline qualtran/bloqs/chemistry/thc_select.py:115]: sorry, can you fix the formatting of these as well

**Review by mpharrigan — COMMENTED** [review]
this generally LGTM, but the description says we have to merge the other one first (?)

**Comment by fdmalone** [comment]
it would be good to generate the notebooks for this too.

File: qualtran/bloqs/chemistry/thc_select.py (hunk 1 above)
Comment (fdmalone) [reply qualtran/bloqs/chemistry/thc_select.py:115]: Done.

**Comment by mpharrigan** [comment]
It's probably better to merge this and add the notebook in a follow-on PR for forward-momentum purposes

**Review by mpharrigan — APPROVED** [review]

**Pull request merged**